camelcase false positive with `global` and `new`

**Tell us about your environment**

* **ESLint Version:** v3.7.1
* **Node Version:** v6.7.0
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```
rules:
  camelcase: 2
```

**What did you do? Please include the actual source code causing the issue.**
```js
/* global My_Global */
new My_Global;
```
or
```js
/* global My_Global */
new Object(My_Global);
```

**What did you expect to happen?**
No error should be given, based on the fact that in a browser, on can usually not change the name of globals provided. Interestingly, the bug only happens when `new` is involved, e.g. the code passes once `new` is removed.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  2:5  error  Identifier 'My_Global' is not in camel case  camelcase
```

--- a/docs/rules/camelcase.md
+++ b/docs/rules/camelcase.md
@@ -52,6 +52,8 @@ var foo = bar.baz_boom;
 var foo = { qux: bar.baz_boom };
 
 obj.do_something();
+do_something();
+new do_something();
 
 var { category_id: category } = query;
 ```

--- a/lib/rules/camelcase.js
+++ b/lib/rules/camelcase.js
@@ -38,6 +38,7 @@ module.exports = {
 
         // contains reported nodes to avoid reporting twice on destructuring with shorthand notation
         const reported = [];
+        const ALLOWED_PARENT_TYPES = new Set(["CallExpression", "NewExpression"]);
 
         /**
          * Checks if a string contains an underscore and isn't all upper-case
@@ -118,7 +119,7 @@ module.exports = {
                         return;
                     }
 
-                    if (isUnderscored(name) && effectiveParent.type !== "CallExpression") {
+                    if (isUnderscored(name) && !ALLOWED_PARENT_TYPES.has(effectiveParent.type)) {
                         report(node);
                     }
 
@@ -131,7 +132,7 @@ module.exports = {
                     }
 
                 // Report anything that is underscored that isn't a CallExpression
-                } else if (isUnderscored(name) && effectiveParent.type !== "CallExpression") {
+                } else if (isUnderscored(name) && !ALLOWED_PARENT_TYPES.has(effectiveParent.type)) {
                     report(node);
                 }
             }

--- a/tests/lib/rules/camelcase.js
+++ b/tests/lib/rules/camelcase.js
@@ -26,6 +26,8 @@ ruleTester.run("camelcase", rule, {
         "myPrivateVariable_ = \"Patrick\"",
         "function doSomething(){}",
         "do_something()",
+        "new do_something",
+        "new do_something()",
         "foo.do_something()",
         "var foo = bar.baz_boom;",
         "var foo = bar.baz_boom.something;",

